src/OmniSharp/MSBuild/ProjectFile/ProjectFileInfo.cs

```diff
@@ -175,5 +180,24 @@ public static ProjectFileInfo Create(MSBuildOptions options, ILogger logger, str
 #endif
             return projectFileInfo;
         }
+        private static LanguageVersion ToLanguageVersion(string langVersionPropertyValue)
+        {
+            if (!(string.IsNullOrWhiteSpace(langVersionPropertyValue) || langVersionPropertyValue.Equals("Default", StringComparison.OrdinalIgnoreCase)))
+            {
+                // ISO-1, ISO-2, 3, 4, 5, 6 or Default
+                switch (langVersionPropertyValue.ToLower())
+                {
+                    case "ISO-1": return LanguageVersion.CSharp1;
+                    case "ISO-2": return LanguageVersion.CSharp2;
+                    case "3": return LanguageVersion.CSharp3;
+                    case "4": return LanguageVersion.CSharp4;
+                    case "5": return LanguageVersion.CSharp5;
+                    case "6": return LanguageVersion.CSharp6;
+                }
+            }
+            return LanguageVersion.CSharp5;
```

Comment:
That sounds like the right thing to do to me. Just don't specify LanguageVersion at all if it's not specified by the csproj file.